src/main/java/org/apache/cassandra/sidecar/exceptions/ThrowableUtils.java

```diff
@@ -101,12 +102,16 @@ public static Throwable getCause(Throwable throwable, Predicate<Throwable> predi
                 return cause;
             }
 
-            fastTracer = getCause(fastTracer, 2);
-            if (cause == fastTracer && stop == null)
+            if (stop == null)
             {
-                // Mark the position to stop, and continue tracing the cause up until hitting stop the next time.
-                // This way we are sure that all exceptions/causes are visited at least once.
-                stop = cause;
+                // once stop is set; updating fast tracer is no longer required
+                if (cause == fastTracer)
+                {
+                    // Mark the position to stop, and continue tracing the cause up until hitting stop the next time.
+                    // This way we are sure that all exceptions/causes are visited at least once.
+                    stop = cause;
+                }
+                fastTracer = getCause(fastTracer, 2);
```

Comment:
it seems we need an else here
```suggestion
                }
                else
                {
                    fastTracer = getCause(fastTracer, 2);
                 }
```